renku/core/models/datasets.py

```diff
@@ -122,7 +133,7 @@ def _convert_creators(value):
 class CreatorMixin:
     """Mixin for handling creators container."""
 
-    creators = jsonld.container.list(Person, kw_only=True, context="schema:creator", converter=_convert_creators)
+    creators = attr.ib(kw_only=True, converter=_convert_creators)
```

Comment:
`factory=list` does not denote this as a list, instead it is almost the same as `default=[]`, but the latter can have issues that the `[]` is the same instance, so if instantiated twice and you add an entry to the first instance, it'll show up on the second instance as well (Which is why we should never use `default=[]` or `default={}` and always use factory instead).

So `factory=list` should only be added if there's not a default like `default=None` already and if the property is not a required property. In this case, I think creators is required, so we don't want a default here. This prevents `CreatorMixin` from being instantiated without passing `creators` to the constructor.

This might be true for the other cases you mention as well, but maybe we should double check.